spec/services/order_builder_spec.rb

```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe PurchasesHandler::OrderBuilder, type: :service do
+
+  describe "#call" do
+
+    context "when order is created" do
```

Comment:
When somente é utilizado para condicionais. Como você não possue nenhuma no código, poderia deixar somente o it.
Exemplo de template:

```ruby
 describe '#nome_metodo_publico ou .nome_metodo_estatico' do
    context 'When detalhes_condicao is true' do
      it 'descricao_it' do
      end
    end

    context 'When detalhes_condicao is false' do
      it 'descricao_it' do
      end
    end
  end
  ```